dll/win32/shell32/folders/CFSFolder.cpp

```diff
@@ -596,13 +596,34 @@ HRESULT SHELL32_GetFSItemAttributes(IShellFolder * psf, LPCITEMIDLIST pidl, LPDW
 
     BOOL bDirectory = (dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
 
+    if (SFGAO_VALIDATE & *pdwAttributes)
+    {
+        STRRET strret;
+        if (SUCCEEDED(psf->GetDisplayNameOf(pidl, SHGDN_FORPARSING, &strret)))
+        {
+            LPWSTR path;
+            if (SUCCEEDED(StrRetToStrW(&strret, pidl, &path)))
+            {
+                BOOL exists = PathFileExistsW(path);
+                SHFree(path);
+                if (!exists)
+                    return E_FAIL;
+            }
+        }
```

Comment:
Could be collapsed a bit more:
```suggestion
        LPWSTR path;
        if (SUCCEEDED(psf->GetDisplayNameOf(pidl, SHGDN_FORPARSING, &strret)) &&
            SUCCEEDED(StrRetToStrW(&strret, pidl, &path)))
        {
            BOOL exists = PathFileExistsW(path);
            SHFree(path);
            if (!exists)
                return E_FAIL;
        }
```